tests/test_generate_dataset_from_htme.py

```diff
@@ -0,0 +1,171 @@
+import ast
+import zlib
+
+import boto3
+import pytest
+from moto import mock_s3
+
+import steps
+from steps import generate_dataset_from_htme
+
+DB_CORE_CONTRACT_FILE_NAME = 'db.core.contract.01002.4040.gz.enc'
+DB_CORE_ACCOUNTS_FILE_NAME = 'db.core.accounts.01002.4040.gz.enc'
+S3_PREFIX = 'mongo/ucdata'
+S3_HTME_BUCKET = 'test'
+S3_PUBLISH_BUCKET = 'target'
+DB_CORE_CONTRACT = 'db.core.contract'
+DB_CORE_ACCOUNTS = 'db.core.accounts'
+SECRETS = "{'collections_all': {'db.core.contract': 'crown'}}"
+SECRETS_COLLECTIONS = {DB_CORE_CONTRACT: 'crown'}
+KEYS_MAP = {"test_ciphertext": "test_key"}
+RUN_TIME_STAMP = "10-10-2000_10-10-10"
+PUBLISHED_DATABASE_NAME = "test_db"
+
+
+def test_retrieve_secrets(monkeypatch, aws_credentials):
+    class MockSession:
+        class Session:
+            def client(self, service_name):
+                class Client:
+                    def get_secret_value(self, SecretId):
+                        return {"SecretBinary": str.encode(SECRETS)}
+
+                return Client()
+
+    monkeypatch.setattr(boto3, 'session', MockSession)
+    assert generate_dataset_from_htme.retrieve_secrets() == ast.literal_eval(SECRETS)
+
+
+def test_get_collections():
+    secret_dict = ast.literal_eval(SECRETS)
+    assert generate_dataset_from_htme.get_collections(secret_dict) == SECRETS_COLLECTIONS
+
+
+@mock_s3
+def test_get_list_keys_for_prefix(aws_credentials):
+    s3_client = boto3.client('s3')
+    s3_client.create_bucket(Bucket=S3_HTME_BUCKET)
+    s3_client.put_object(Body="test", Bucket=S3_HTME_BUCKET, Key=f'{S3_PREFIX}/{DB_CORE_CONTRACT_FILE_NAME}')
+
+    assert (
+            generate_dataset_from_htme.get_list_keys_for_prefix(s3_client, S3_HTME_BUCKET, S3_PREFIX) ==
+            [f"{S3_PREFIX}/{DB_CORE_CONTRACT_FILE_NAME}"]
+    )
+
+
+def test_group_keys_by_collection():
+    keys = [f'{S3_PREFIX}/{DB_CORE_CONTRACT_FILE_NAME}',
+            f'{S3_PREFIX}/db.core.accounts.01002.4040.gz.enc']
+    assert (
+            generate_dataset_from_htme.group_keys_by_collection(keys) ==
+            [{f'{DB_CORE_CONTRACT}': [f'{S3_PREFIX}/{DB_CORE_CONTRACT_FILE_NAME}']},
+             {'db.core.accounts': [f'{S3_PREFIX}/db.core.accounts.01002.4040.gz.enc']}]
```

Comment:
This constant DB_CORE_ACCOUNTS_FILE_NAME can be leveraged here ?

Reply:
good point i will pick this up on Monday as I have to change my local git config as well as Dan Hill pointed out to resolve my details linked with the commit